### ISSUE TITLE
Can't register new domain - azurerm_app_service_domain needed

### ISSUE THREAD

**Issue by holmesb** [terraform-providers/terraform-provider-azurerm#4683]
I can register a new Internet domain using the App Service Domains service in the Portal --> Create App Service Domain.

I see no way to do the same using this provider, hence I expect this is a feature request for a azurerm_app_service_domain resource & data source.  This is non-trivial because integration testing during development will cost $$, deletion will be just disabling auto-renewal, etc.  Seems this feature will land shortly in the AWS provider, and can follow much of [their thought process here](https://github.com/terraform-providers/terraform-provider-aws/issues/88#issuecomment-524815670).

**Comment by magodo** [comment]
@holmesb I tried to implement this resource, but get blocked for several reasons. I have opened an upstream issue: Azure/azure-rest-api-specs#10223 to track the blockers.

**Comment by zadigus** [comment]
what is the status of this feature?

**Comment by techadrian** [comment]
Guys, I might be wrong, but missing this azurerm_app_service_domain  feature makes programmatical public domain registration with terraform in azure impossible.  Any updates regarding this feature?

**Comment by techadrian** [comment]
> 

Thank you so much for sharing this "workaround". 
I used the local-exec provisioner to execute "az appservice domain create" commands with terraform but using azapi_resource is far way better.

**Comment by zadigus** [comment]
@techadrian yeah, it's not optimal, but at least, you get reliable terraform support for that resource; I hate the local-exec provisioner way, because it makes it very hard to track resources properly (e.g. for updates).
